bug: algorithm crash in `rank_soils` when passing in empty data

## Description
Rank soils crashes when provided with empty data. For example,
```
    rank_soils(
        lon=-122.126583,
        lat=37.430296,
        list_output_data=list_soils(-122.126583, 37.430296),
        soilHorizon=[],
        horizonDepth=[],
        rfvDepth=[],
        lab_Color=[],
        pSlope=None,
        pElev=None,
        bedrock=None,
        cracks=None,
    )
```

## Steps To Reproduce
<!--
Steps to reproduce the behavior. Describe what you typed, clicked, tapped, swiped, saw and heard.
-->
1. Go to URL: [insert URL here]
2. <!-- Click on '....' -->
3. <!-- Scroll down to '....' -->
4. <!-- See error -->

## Expected behavior
No crash.

## Actual behavior
Following traceback:
```
 # Check if user has entered horizon data
>       if p_hz_data.empty or p_bottom_depth.bottom_depth.le(0).any():
E       AttributeError: 'NoneType' object has no attribute 'empty'

soil_id/us_soil.py:1750: AttributeError
```

## Additional context
I think this is an easy fix.

--- a/soil_id/tests/us/test_us.py
+++ b/soil_id/tests/us/test_us.py
@@ -3,22 +3,22 @@ import time
 
 from soil_id.us_soil import list_soils, rank_soils
 
+test_locations = [
+    {"lon": -101.9733687, "lat": 33.81246789},
+    {"lon": -121.0347381, "lat": 45.88932423},
+    {"lon": -85.50621214, "lat": 39.26009312},
+    {"lon": -94.31005777, "lat": 42.63413723},
+    {"lon": -99.55016693, "lat": 37.48216451},
+    {"lon": -157.2767099, "lat": 62.32776717},
+    {"lon": -156.4422738, "lat": 63.52666854},
+    {"lon": -119.4596489, "lat": 43.06450312},
+    {"lon": -69.28246582, "lat": 47.21392200},
+    {"lon": -158.4018264, "lat": 60.42282639},
+    {"lon": -121.8166, "lat": 48.6956},
+]
+
 
 def test_soil_location():
-    test_locations = [
-        {"lon": -101.9733687, "lat": 33.81246789},
-        {"lon": -121.0347381, "lat": 45.88932423},
-        {"lon": -85.50621214, "lat": 39.26009312},
-        {"lon": -94.31005777, "lat": 42.63413723},
-        {"lon": -99.55016693, "lat": 37.48216451},
-        {"lon": -157.2767099, "lat": 62.32776717},
-        {"lon": -156.4422738, "lat": 63.52666854},
-        {"lon": -119.4596489, "lat": 43.06450312},
-        {"lon": -69.28246582, "lat": 47.21392200},
-        {"lon": -158.4018264, "lat": 60.42282639},
-        {"lon": -121.8166, "lat": 48.6956},
-    ]
-
     # Dummy Soil Profile Data (replicating the structure provided)
     soilHorizon = ["LOAM"] * 7
     horizonDepth = [1, 10, 20, 50, 70, 100, 120]
@@ -49,3 +49,20 @@ def test_soil_location():
         )
         print(list_soils_result)
         print(rank_soils_result)
+
+
+def test_empty_rank():
+    list_result = list_soils(test_locations[0]["lon"], test_locations[0]["lat"])
+    rank_soils(
+        lon=test_locations[0]["lon"],
+        lat=test_locations[0]["lat"],
+        list_output_data=list_result,
+        soilHorizon=[],
+        horizonDepth=[],
+        rfvDepth=[],
+        lab_Color=[],
+        pSlope=None,
+        pElev=None,
+        bedrock=None,
+        cracks=None,
+    )

--- a/soil_id/us_soil.py
+++ b/soil_id/us_soil.py
@@ -1747,7 +1747,7 @@ def rank_soils(
             soil_matrix.iloc[:, i] = 1
 
     # Check if user has entered horizon data
-    if p_hz_data.empty or p_bottom_depth.bottom_depth.le(0).any():
+    if p_hz_data is None or p_hz_data.empty or p_bottom_depth.bottom_depth.le(0).any():
         soilIDRank_output_pd = None
     else:
         # Subset component soil properties to match user measured properties

